app.go

```diff
@@ -82,7 +82,7 @@ you can use to register schedules and listeners.
 */
 func NewApp(request NewAppRequest) (*App, error) {
 	if request.IpAddress == "" || request.HAAuthToken == "" || request.HomeZoneEntityId == "" {
-		log.Fatalln("IpAddress, HAAuthToken, and HomeZoneEntityId are all required arguments in NewAppRequest.")
+		slog.Error("IpAddress, HAAuthToken, and HomeZoneEntityId are all required arguments in NewAppRequest.")
```

Comment:
does `slog.Error` also exit like `log.Fatalln` does?

Reply:
Good point, slog.Error() doesn't exit execution. 
After reading a few discussions and recommendations in books, my summary of the idiomatic way to propagate similar unexpected situations from libraries should still be using Errors (not Fatal or panic). E.g., it allows gracefully catching this in unit/integration tests.
I fixed it according to it.

Reply:
I agree that is idiomatic in Go, but if there's an error that will prevent GA from working at all I didn't want less experienced Go users to be left wondering why their program was exiting immediately.

However you're probably right that it's best to just follow convention. This solution is good for now, thanks!